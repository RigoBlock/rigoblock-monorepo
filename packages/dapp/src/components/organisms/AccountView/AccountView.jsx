import './AccountView.scss'
import { connect } from 'react-redux'
import AccountItem from '../../molecules/AccountItem'
import List from '../List'
import PropTypes from 'prop-types'
import React from 'react'
import Title, { TITLE_SIZES } from '../../atoms/Title'

let AccountView = ({ currentAccount, accounts }) => {
  const accountList = Object.keys(accounts).map(account => ({
    number: account,
    provider: accounts[account].provider
  }))
  return currentAccount ? (
    <div className="account-view">
      <Title size={TITLE_SIZES.SMALL}>Accounts</Title>
      <List
        Component={AccountItem}
        data={accountList}
        className={'account-list'}
      />
    </div>
  ) : (
    <div className="account-view">
      <h1>Accounts</h1>
      <div className="typeface">¯\_(ツ)_/¯</div>
    </div>
  )
}

AccountView.propTypes = {
  accounts: PropTypes.object.isRequired,
  currentAccount: PropTypes.string
}

AccountView.defaultProps = {
  currentAccount: null
}

AccountView = connect(state => ({
  currentAccount: state.preferences.currentAccount,
  accounts: state.blockChain.accounts
}))(AccountView)

export default AccountView
