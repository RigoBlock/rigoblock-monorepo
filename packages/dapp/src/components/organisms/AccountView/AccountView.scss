@import "../../_settings/variables.scss";

.account-view {
  h1 {
    font-size: 20px;
    font-weight: 700px;
  }

  .account-list {
    :first-child {
      padding-top: $dist-2xs;
    }

    > *:not(:first-child) {
      padding-top: $dist-s;
    }
  }

  .typeface {
    padding-top: $dist-xs;
    font-family: Menlo;
  }
}
