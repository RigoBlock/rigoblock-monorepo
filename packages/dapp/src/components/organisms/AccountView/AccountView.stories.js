import '../../_settings/_base.scss'
import { Provider } from 'react-redux'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import AccountView from './AccountView'
import React from 'react'
import mockStore from '../../../fixtures/store'

const store = mockStore()

storiesOf('Organisms/AccountView', module)
  .addDecorator(withKnobs)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .add('default', () => <AccountView />)
