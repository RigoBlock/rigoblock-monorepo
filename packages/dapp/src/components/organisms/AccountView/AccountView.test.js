import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import AccountView from './AccountView.jsx'
import React from 'react'
import toJson from 'enzyme-to-json'

const mockStore = {
  getState: () => ({
    user: {
      blockChain: {
        accounts: {
          '0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196': {
            provider: 'metamask'
          },
          '0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d': {
            provider: 'metamask'
          },
          '0x8bb7481495d45ccd5cffae1c3a84155fea85a323': {
            provider: 'metamask'
          },
          '0xe198d98b76c529886affb5a74be8b435624bd310': {
            provider: 'metamask'
          },
          '0x927aa991a628aab6851a890fb790a4a7cd0ec446': {
            provider: 'metamask'
          }
        }
      },
      preferences: {
        currentAccount: '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196'
      }
    }
  }),
  dispatch: jest.fn(),
  subscribe: () => null
}

const wrapper = mount(
  <Provider store={mockStore}>
    <AccountView />
  </Provider>
)

describe('PreferencesForm component', () => {
  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
