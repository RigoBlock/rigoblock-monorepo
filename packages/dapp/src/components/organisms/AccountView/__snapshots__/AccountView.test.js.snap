// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreferencesForm component renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AccountView)>
    <AccountView
      accounts={
        Object {
          "0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196": Object {
            "provider": "metamask",
          },
          "0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d": Object {
            "provider": "metamask",
          },
          "0x8bb7481495d45ccd5cffae1c3a84155fea85a323": Object {
            "provider": "metamask",
          },
          "0x927aa991a628aab6851a890fb790a4a7cd0ec446": Object {
            "provider": "metamask",
          },
          "0xe198d98b76c529886affb5a74be8b435624bd310": Object {
            "provider": "metamask",
          },
        }
      }
      currentAccount="0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196"
      dispatch={[MockFunction]}
    >
      <div
        className="account-view"
      >
        <Title
          className={null}
          size="small"
        >
          <div
            className="title small"
          >
            Accounts
          </div>
        </Title>
        <List
          Component={[Function]}
          className="account-list"
          data={
            Array [
              Object {
                "number": "0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196",
                "provider": "metamask",
              },
              Object {
                "number": "0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d",
                "provider": "metamask",
              },
              Object {
                "number": "0x8bb7481495d45ccd5cffae1c3a84155fea85a323",
                "provider": "metamask",
              },
              Object {
                "number": "0xe198d98b76c529886affb5a74be8b435624bd310",
                "provider": "metamask",
              },
              Object {
                "number": "0x927aa991a628aab6851a890fb790a4a7cd0ec446",
                "provider": "metamask",
              },
            ]
          }
        >
          <div
            className="account-list"
          >
            <AccountItem
              key="0"
              number="0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196"
              provider="metamask"
            >
              <div
                className="account-item"
              >
                <div
                  className="account-item-logo"
                >
                  <Component>
                    SVG image metamask-fox.svg
                  </Component>
                </div>
                <div
                  className="account-item-info"
                >
                  <span
                    className="account-item-provider"
                  >
                    MetaMask
                  </span>
                  <span
                    className="account-item-number"
                  >
                    **** 9196
                  </span>
                </div>
              </div>
            </AccountItem>
            <AccountItem
              key="1"
              number="0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d"
              provider="metamask"
            >
              <div
                className="account-item"
              >
                <div
                  className="account-item-logo"
                >
                  <Component>
                    SVG image metamask-fox.svg
                  </Component>
                </div>
                <div
                  className="account-item-info"
                >
                  <span
                    className="account-item-provider"
                  >
                    MetaMask
                  </span>
                  <span
                    className="account-item-number"
                  >
                    **** 2d6d
                  </span>
                </div>
              </div>
            </AccountItem>
            <AccountItem
              key="2"
              number="0x8bb7481495d45ccd5cffae1c3a84155fea85a323"
              provider="metamask"
            >
              <div
                className="account-item"
              >
                <div
                  className="account-item-logo"
                >
                  <Component>
                    SVG image metamask-fox.svg
                  </Component>
                </div>
                <div
                  className="account-item-info"
                >
                  <span
                    className="account-item-provider"
                  >
                    MetaMask
                  </span>
                  <span
                    className="account-item-number"
                  >
                    **** a323
                  </span>
                </div>
              </div>
            </AccountItem>
            <AccountItem
              key="3"
              number="0xe198d98b76c529886affb5a74be8b435624bd310"
              provider="metamask"
            >
              <div
                className="account-item"
              >
                <div
                  className="account-item-logo"
                >
                  <Component>
                    SVG image metamask-fox.svg
                  </Component>
                </div>
                <div
                  className="account-item-info"
                >
                  <span
                    className="account-item-provider"
                  >
                    MetaMask
                  </span>
                  <span
                    className="account-item-number"
                  >
                    **** d310
                  </span>
                </div>
              </div>
            </AccountItem>
            <AccountItem
              key="4"
              number="0x927aa991a628aab6851a890fb790a4a7cd0ec446"
              provider="metamask"
            >
              <div
                className="account-item"
              >
                <div
                  className="account-item-logo"
                >
                  <Component>
                    SVG image metamask-fox.svg
                  </Component>
                </div>
                <div
                  className="account-item-info"
                >
                  <span
                    className="account-item-provider"
                  >
                    MetaMask
                  </span>
                  <span
                    className="account-item-number"
                  >
                    **** c446
                  </span>
                </div>
              </div>
            </AccountItem>
          </div>
        </List>
      </div>
    </AccountView>
  </Connect(AccountView)>
</Provider>
`;
